File: skills/scaffold-umbrella-with-turbo-and-pnpm/assets/templates/recipes/kmp/shared/src/commonMain/kotlin/Calculator.kt
package com.umbrella.example

class Calculator {
    fun add(left: Double, right: Double): Double = left + right

    fun subtract(left: Double, right: Double): Double = left - right

    fun multiply(left: Double, right: Double): Double = left * right

    fun divide(left: Double, right: Double): Double {
        require(right != 0.0) { "Cannot divide by zero" }
        return left / right
    }
}

File: skills/scaffold-umbrella-with-turbo-and-pnpm/assets/templates/packages/typescript-calculator/src/calculator.ts
export class Calculator {
  add(left: number, right: number): number {
    return left + right;
  }

  subtract(left: number, right: number): number {
    return left - right;
  }

  multiply(left: number, right: number): number {
    return left * right;
  }

  divide(left: number, right: number): number {
    if (right === 0) {
      throw new RangeError('Cannot divide by zero');
    }

    return left / right;
  }
}
